pkg/manager/internal.go

```diff
@@ -623,13 +644,34 @@ func (cm *controllerManager) startLeaderElection(ctx context.Context) (err error
 				if cm.onStoppedLeading != nil {
 					cm.onStoppedLeading()
 				}
+				var retErrors = []error{
+					errors.New("leader election lost"),
+				}
+
+				for _, cb := range cm.onStoppedLeadingCallback {
+					if err := cb(); err != nil {
+						retErrors = append(retErrors, err)
```

Comment:
1. What's this error for? Even we put it into `errChan`, it will only be logged for the Start error. Maybe just let the callback itself print the error log?
2. Can we just reuse `cm.onStoppedLeading` since it is only used for test now?

Reply:
1. //ACK. Sounds good to me. The idea however was to be able to aggregate and display all the errors in one place with regards to handling the leader election lost item. But I have no issue with callback handlers logging it and forgetting about it instead. 
2. Due to https://github.com/kubernetes-sigs/controller-runtime/pull/1919#discussion_r885550142 I decided to keep the callback for stopped leading in a different attribute.